src/swell/tasks/link_coupled_geos_output.py

```diff
@@ -0,0 +1,252 @@
+        duration = isodate.parse_duration(background_frequency)
+        hist_hours, remainder = divmod(duration.total_seconds(), 60*60)
+
+        return f'{int(hist_hours):02d}h'
+
+    # ----------------------------------------------------------------------------------------------
+
+    def link_cice6_history_4d(self) -> None:
+        # Creating links to GEOS history, CICE6, for SOCA inputs
+        # Depending on the DA type, there could be multiple state files to link
+        # -----------------------------------------------------------------------
+        states = self.geos.states_generator(self.background_frequency, self.window_length,
+                                            self.window_begin_iso, self.get_model(),
+                                            self.marine_models)
+
+        hour_prefix = self.cice6_history_hour_prefix(self.background_frequency)
+
+        # Get date information from the states dictionary, and use ocn_filename to get the
+        # destination file name
+        for state in states:
+            src_date = dt.strptime(state['date'], datetime_formats['iso_format'])
+            src_history = self.cice6_history_formatter(src_date, hour_prefix)
+            dst_history = os.path.join(self.cycle_dir(), state['ice_filename'])
+            self.prepare_cice6_history(src_history, dst_history)
+
+        # Using a 4D window; CICE6 background is at the beginning of the window
+        src_history = self.cice6_history_formatter(self.bkgr_time_dto, hour_prefix)
+        dst_history = os.path.join(self.cycle_dir(), 'cice.res.' + self.bkgr_time_iso + '.nc')
+
+        self.prepare_cice6_history(src_history, dst_history)
+
+    # ----------------------------------------------------------------------------------------------
+
+    def link_cice6_history_3d(self) -> None:
+        # Using a 3D window; CICE6 background is in the middle of the window
+        hour_prefix = self.cice6_history_hour_prefix()
+        src_history = self.cice6_history_formatter(self.bkgr_time_dto, hour_prefix)
+        dst_history = os.path.join(self.cycle_dir(), 'cice.res.' + self.bkgr_time_iso + '.nc')
+
+        self.prepare_cice6_history(src_history, dst_history)
+
+    # ----------------------------------------------------------------------------------------------
+
+    def prepare_cice6_history(self,
+                              src_history: str,
+                              dst_history: str,
+                              ) -> None:
+
+        # Since history already has the aggregated variables, we just need to rename
+        # the dimensions and variables to match SOCA requirements
+        ds = xr.open_dataset(src_history)
+
+        # rename the dimensions to xaxis_1 and yaxis_1 and rename the variables
+        ds = ds.rename({'ni': 'xaxis_1', 'nj': 'yaxis_1'})
+        ds = ds.rename({'aice': 'aice_h', 'hi': 'hi_h', 'hs': 'hs_h'})
+
+        # Save as a new file
+        ds.to_netcdf(dst_history, mode='w')
```

Comment:
Suggest being explicit about format here.

```suggestion
        ds.to_netcdf(dst_history, mode='w', format='NETCDF4')
```

Reply:
This would write a *.nc4 file, correct?

Reply:
`format="NETCDF4"` has nothing to do with the file extension. It will write to whatever the full value of `dst_history` is. If `dst_history = "somefile.crazy"`, it will write a file called `somefile.crazy`.

The point here is to (1) make it clear to other/future programmers that we're writing NetCDF4 files; and (2) make this code agnostic to the current dependencies installed and to `xarray.to_netcdf`'s internal rules about how it decides what kinds of "netcdf" files to write by default (e.g., right now, I think the rule is something like: Use NetCDF4 if h5netcdf or netcdf4 is installed; otherwise, fall back to netcdf3...but that might change in the future).